### DISCUSSION TITLE
Question about UI

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#16606] [Q&A] [answered]
I'm learning Avalonia UI and recently found out that there is Jetpack Compose for Kotlin and that for Kotlin developers are learning it. Now I'm afraid that there is something similar for Avalonia and that I'm behind the times? Is this true or can I just learn Axaml and not think about it?I'm learning Avalonia UI and recently found out that there is Jetpack Compose for Kotlin and that for Kotlin developers are learning it. Now I'm afraid that there is something similar for Avalonia and that I'm behind the times? Is this true or can I just learn Axaml and not think about it?

**Comment by thevortexcloud** [comment, marked as answer]
Avalonia is a .NET (primarily C#) GUI framework. Kotlin/Jetpack compose are different things entirely. There is some overlap between them and Avalonia though in terms of what they do.

As far as I am aware, there is no plans to replace XAML in Avalonia. The XAML community in general is also not going anywhere. WPF is still heavily used by developers despite it not being the "reccomended" framework by Microsoft for a long time now. Once you learn one XAML flavour you can easily learn any of them. So the skills you learn from Avalonia can easily be transferred even if Avalonia moves away from XAML.

I could however have dreamt this discussion discussion: But I seem to recall someone saying Avalonia does technically support additional markup languages and is not strongly tied to XAML. But nobody has added any. 

EDIT: Found the discussion

https://github.com/AvaloniaUI/Avalonia/discussions/14255

**Reply by al-zakaria** [reply]
by the way the markup in this discussion looks very decent, but the community doesn't want to go for this type of markup? if so then I'm calm

**Reply by thevortexcloud** [reply]
It's unlikely to happen any time soon, at least officially. Most people are using Avalonia because it has XAML, which makes it easy to transition from things like WPF.

**Reply by al-zakaria** [reply]
thank you very much
